src/PostgresWebsockets.hs

```diff
@@ -81,11 +81,12 @@ wsApp getTime dbChannel secret pool multi pendingConn =
               Nothing -> pure ()
 
             when (hasRead mode) $
-              onMessage multi ch $ WS.sendTextData conn . B.payload
+                flip (onMessage multi) (WS.sendTextData conn . B.payload) <$> chs
+--              onMessage multi ch $ WS.sendTextData conn . B.payload
 
             when (hasWrite mode) $
               let sendNotifications = void . H.notifyPool pool dbChannel . toS
-              in notifySession validClaims (toS ch) conn getTime sendNotifications
+              in notifySession validClaims (toS chs) conn getTime sendNotifications
```

Comment:
you will need to align the `chs` type with `notifySession`. Since `notifySession` is expecting a `Text` here this is messing the type inference for `chs`. We need a similar approach to the `onMessage` call using `forM_`.
Instead of using flip here I recommend just rewriting the `notifySession` seignature to leave the channel argument in the last position.

Reply:
I have attempted to do this in 0c5d84e9140d75273dc93fed03fa06a22344b4db

Reply:
not very well though, as I just use the first channel